cql3/statements/drop_table_statement.cc

```diff
@@ -76,18 +80,24 @@ future<shared_ptr<cql_transport::event::schema_change>> drop_table_statement::an
 {
     return make_ready_future<>().then([this, is_local_only] {
         return service::get_local_migration_manager().announce_column_family_drop(keyspace(), column_family(), is_local_only);
-    }).then_wrapped([this] (auto&& f) {
+    }).then_wrapped([this, &proxy] (auto&& f) {
         try {
             f.get();
             using namespace cql_transport;
-            return make_shared<event::schema_change>(
-                    event::schema_change::change_type::DROPPED,
-                    event::schema_change::target_type::TABLE,
-                    this->keyspace(),
-                    this->column_family());
+            auto fut = make_ready_future<>();
+            if (proxy.get_db().local().has_schema(keyspace(), cdc::desc_name(column_family()))) {
```

Comment:
Could it happen that `proxy.get_db().local().has_schema(keyspace(), cdc::desc_name(column_family()))` returns false on our shard/node, but returns true on a different shard/node, because the information about the cdc table didn't yet arrive to our shard?

Wouldn't unconditional cdc::remove be better? Announce the remove to each node/shard, and simply make it noop if the table doesn't exist locally.

Reply:
unconditional remove is probably better. Will change.